movement/watch_faces/complication/totp_face_lfs.c

```diff
@@ -279,15 +286,37 @@ static void totp_face_display(totp_lfs_state_t *totp_state) {
     watch_display_string(buf, 0);
 }
 
+static void totp_face_lfs_switch_to_secmod(totp_lfs_state_t *totp_state) {
+    int i;
+
+    for (i = 0; i < num_totp_records; ++i) {
+        totp_face_lfs_get_file_secret(&totp_records[i]);
+        totp_records[i].secret = malloc(totp_records[i].secret_size);
+        if (totp_records[i].secret == NULL) {
+            num_totp_records = i - 1;
+            // We can't easily undo at this point, because we destroyed
+            // the offset info (secret is in a union). So let's just junk
+            // the remaining records.
+            break;
+        }
+        memcpy(totp_records[i].secret, current_secret, totp_records[i].secret_size);
+    }
+
+    filesystem_rm(TOTP_FILE);
```

Comment:
This is unlikely to be a secure erase operation. For performance reasons, file systems typically don't overwrite file contents when erasing, they just remove the links to the file from the file system data structures, causing the file to become unreachable and the memory to become available for new files. The data will not be overwritten until new data is written to the same physical location.

For this reason, it may be wise to overwrite the entire file with zeros before issuing the `rm` command to the file system.

This is adequate for spinning disks but may not be completely effective for flash storage: copies of the file may still linger on the storage medium if wear leveling is employed.

I do not know if the hardware supports a secure erase operation but it's probably a good idea to _also_ issue those commands if it does. Even if it does, it is unknown how effective it is, hardware manufacturers are known for screwing up features like this. The data might therefore still be recoverable despite all possible precautions.

Reply:
:+1: 

Yes, as I mentioned at the top, don't know enough about LFS to understand exactly what will happen here. I wouldn't feel confident in zero-ing it out as a method because I haven't looked at LFS's internal structure (I mean, if it was the same length... but as you mention, LFS could have some built-in wear levelling that means it will write it in a separate block). At least there's no hardware level magic here we have to worry about!

I'll change the top of the PR to make these caveats clearer.